### PR TITLE
Make OxyPlot structs immutable

### PR DESCRIPTION
**Pull request by objorke**
For #692

### CODE REVIEW

**Comment by objorke** [comment]
@tibel can you review this PR?

File: Source/OxyPlot/Rendering/OxySize.cs [hunk 1]
```diff
@@ -31,21 +41,33 @@ public struct OxySize : IFormattable
         public OxySize(double width, double height)
             : this()
```
Comment (tibel) [inline Source/OxyPlot/Rendering/OxySize.cs]: I think calling a default constructor on a struct is not needed.
All memory is zeroed by default.

**Comment by tibel** [comment]
Looks good. 

Maybe adding some helper methods/operators to make code simpler:
e.g. OxyRect add/subtract operator for size
or similar so you don't have to call the constructor with all parameters again

**Comment by objorke** [comment]
I have fixed the OxySize ctor.

File: Source/OxyPlot/Rendering/OxySize.cs (hunk 1 above)
Comment (objorke) [inline Source/OxyPlot/Rendering/OxySize.cs]: Fixed!

**Comment by objorke** [comment]
Yes, I noticed changes to `OxyRect` instances became more complicated.
I prefer `Inflate` and `Offset` methods over operators, as in
https://msdn.microsoft.com/en-us/library/system.windows.rect(v=vs.110).aspx

**Comment by objorke** [comment]
I added some methods to OxyRect, but I am not very happy with the result...
The Inflate/Deflate methods are ok, but the others don't help much...

**Comment by objorke** [comment]
I removed the confusing "change size" methods...

**Comment by tibel** [comment]
Great. Thanks for changing this.

**Pull request merged**